sentinelsat/sentinel.py

```diff
@@ -318,28 +315,42 @@ def download(self, id, directory_path='.', checksum=False, check_existing=False)
 
         self.logger.info('Downloading %s to %s' % (id, path))
 
-        # Check if the file exists and passes md5 test
-        # The download function will by default continue the download if the file exists but is incomplete
-        if exists(path) and getsize(path) == product_info['size']:
-            if not check_existing or _md5_compare(path, product_info['md5']):
-                self.logger.info('%s was already downloaded.' % path)
-                return product_info
+        if exists(path):
```

Comment:
I see your point in removing it. How about another option:

- set `checksum=True` as default und move it up into `if exists(path)`. Use it as a single option to check temp as well as completed files.

You already proposed in another PR that setting `checksum=True` is a sensible default - and I agree. We can use this option to cover current as well as past downloads. Only downside I see is a use case where someone periodically runs the same, large query with the goal to only download new products. Then however they could adapt the query or just use the `limit` keyword.